packages/broker/src/plugins/operator/MaintainOperatorValueHelper.ts

```diff
@@ -1,33 +1,117 @@
+     */
+    async getUnwithdrawnEarningsOf(operatorContractAddress: EthereumAddress): Promise<UnwithdrawnEarningsData> {
+        const operator = new Contract(operatorContractAddress, operatorABI, this.config.signer) as unknown as Operator
+        const minSponsorshipEarningsInWithdrawWei = BigNumber.from(this.config.minSponsorshipEarningsInWithdraw ?? 0)
+        const { sponsorshipAddresses: allSponsorshipAddresses, earnings } = await operator.getEarningsFromSponsorships()
+
+        const sponsorships = allSponsorshipAddresses
+            .map((address, i) => ({ address, earnings: earnings[i] }))
+            .filter((sponsorship) => sponsorship.earnings.gte(minSponsorshipEarningsInWithdrawWei))
+            .sort((a, b) => Number(b.earnings.sub(a.earnings).toBigInt())) // TODO: after Node 20, use .toSorted() instead
+            .slice(0, this.config.maxSponsorshipsInWithdraw) // take all if maxSponsorshipsInWithdraw is undefined
+        const sponsorshipAddresses = sponsorships.map((sponsorship) => toEthereumAddress(sponsorship.address))
+
+        const approxPoolValue = (await operator.totalValueInSponsorshipsWei()).toBigInt()
+        const sumDataWei = sponsorships.reduce((sum, sponsorship) => sum.add(sponsorship.earnings), BigNumber.from(0)).toBigInt()
+        const fraction = approxPoolValue > 0
+            ? sumDataWei * ONE_ETHER / approxPoolValue
+            : BigInt(0)
+
+        return { sumDataWei, fraction, sponsorshipAddresses }
+    }
+
+    async getMyUnwithdrawnEarnings(): Promise<UnwithdrawnEarningsData> {
+        return this.getUnwithdrawnEarningsOf(this.config.operatorContractAddress)
+    }
+
+    async withdrawMyEarningsFromSponsorships(sponsorshipAddresses: EthereumAddress[]): Promise<void> {
+        await (await this.operator.withdrawEarningsFromSponsorships(sponsorshipAddresses)).wait()
+    }
+
+    async triggerWithdraw(targetOperatorAddress: EthereumAddress, sponsorshipAddresses: EthereumAddress[]): Promise<void> {
+        await (await this.operator.triggerAnotherOperatorWithdraw(targetOperatorAddress, sponsorshipAddresses)).wait()
     }
 
-    async updateApproximatePoolValueOfSponsorships(sponsorshipAddresses: string[]): Promise<void> {
-        await (await this.operator.updateApproximatePoolvalueOfSponsorships(sponsorshipAddresses)).wait()
+    private async getOperatorAddresses(requiredBlockNumber: number): Promise<EthereumAddress[]> {
+        const createQuery = () => {
+            return {
+                query: `
+                    {
+                        operators {
+                            id
+                        }
+                    }
+                    `
+            }
+        }
+        const parseItems = (response: { operators: { id: string }[] }) => response.operators
+        this.theGraphClient.updateRequiredBlockNumber(requiredBlockNumber)
+        const queryResult = this.theGraphClient.queryEntities<{ id: string }>(createQuery, parseItems)
```

Comment:
The default parseItems works ok as there is single root level item which is an array:
https://github.com/streamr-dev/network/blob/NET-537_operator_value_breach_watcher/packages/utils/src/TheGraphClient.ts#L61-L62

```suggestion
        const queryResult = this.theGraphClient.queryEntities<{ id: string }>(createQuery)
```

Reply:
Cool, thanks for pointing that out